src/Maestro/Maestro.Web/Startup.cs

```diff
@@ -45,14 +49,89 @@ static Startup()
         {
             Triggers<BuildChannel>.Inserted += entry =>
             {
-                BuildChannel entity = entry.Entity;
                 DbContext context = entry.Context;
+                BuildChannel entity = entry.Entity;
+
+                StartAssociatedReleasePipelinesAsync(entry).GetAwaiter().GetResult();
+
                 var queue = context.GetService<BackgroundQueue>();
                 var dependencyUpdater = context.GetService<IDependencyUpdater>();
                 queue.Post(() => dependencyUpdater.StartUpdateDependenciesAsync(entity.BuildId, entity.ChannelId));
             };
         }
 
+        private static async Task StartAssociatedReleasePipelinesAsync(IInsertedEntry<BuildChannel, DbContext> entry)
+        {
+            BuildAssetRegistryContext context = entry.Context as BuildAssetRegistryContext;
+            ILogger<BuildChannel> logger = context.GetService<ILogger<BuildChannel>>();
+            BuildChannel entity = entry.Entity;
+            Build barBuildInfo = entity.Build;
+            Channel channel = context.Channels
+                .Where(ch => ch.Id == entity.ChannelId)
+                .Include(ch => ch.ChannelReleasePipelines)
+                .ThenInclude(crp => crp.ReleasePipeline)
+                .First();
+
+            // If something use the old API version we won't have this information available.
+            // This will also be the case if something adds an existing build (created using 
+            // the old API version) to a channel
+            if (barBuildInfo.AzureDevOpsBuildId == null)
+            {
+                logger.LogWarning($"barBuildInfo.AzureDevOpsBuildId is null for BAR Build.Id {barBuildInfo.Id}.");
+                return;
+            }
+
+            if (channel.ChannelReleasePipelines?.Any() != true)
```

Comment:
`if (!channel.ChannelReleasePipelines?.Any())`

Reply:
I'm not sure but will "!" convert "null" to false?

Reply:
It might? Either it's probably fine.

Reply:
I'm pretty sure the ! operator will lift, leaving you with still nullable thing.  but it might be clearer just to separate out the null check, so the mind bending of what happens if it's null, or empty, and what does != true do for null.